### PR TITLE
Add docker login step only

### PR DESCRIPTION
**Pull request by JanakaSandaruwan**
https://github.com/wso2-enterprise/choreo/issues/27561

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -5,8 +5,22 @@ const fs = require("fs");
...
 async function run() {
+  switch (type) {
+    case "login_and_push":
+      login_and_push();
+      break;
+    case "login":
+      login();
+      break;
+    default:
+      throw new Error(`Unknown action type: ${type}`);
```
Comment (yashodgayashan) [inline index.js]: shall we have the default behaviour as login and push?

File: index.js [hunk 2]
```diff
@@ -147,6 +190,43 @@ async function dockerHubLogin(cred) {
   }
 }
 
+async function gcpArtifactRegistryLogin(cred) {
```
Comment (yashodgayashan) [inline index.js:193]: I am thinking about whether we can reuse this?

**Review by yashodgayashan — CHANGES_REQUESTED** [review]

File: index.js (hunk 1 above)
Comment (JanakaSandaruwan) [reply index.js]: Fixed with https://github.com/choreo-templates/choreo-image-push/pull/6/commits/c8df47ff2f34384ed6e0b9582901d1b0a6a852e6

File: index.js (hunk 2 above)
Comment (JanakaSandaruwan) [reply index.js:193]: For other registry types, we have two separate functions for login and push. Hence, I added separate one for gcp as well

Comment (JanakaSandaruwan) [reply index.js:193]: This will be tracked via https://github.com/wso2-enterprise/choreo/issues/27820

**Review by yashodgayashan — APPROVED** [review]

**Pull request merged**